AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -265,13 +270,15 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
                        writeOptions:(MSWriteOptions *)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
   // TODO consume writeOptions
-  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
-                   partitions:@[ partition ]
-            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+    [MSTokenExchange
+     performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
+     partition:partition
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull tokenExchangeError) {
               // If error getting token.
-              if (error || !tokenResponses) {
-                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
-                completionHandler([[MSDataSourceError alloc] initWithError:error]);
+              if (tokenExchangeError || !tokenResponses || [tokenResponses.tokens count] <= 0) {
+                  NSNumber *httpErrorCode = [tokenExchangeError userInfo][kMSCosmosDbHttpCodeKey];
+                  MSLogError([MSDataStore logTag], @"Can't get CosmosDb token. Http error code: %ld; Partition: %@", (long)[httpErrorCode integerValue], partition);
```

Comment:
Why casting http status code to `long`? Can't you use `[httpErrorCode intValue]` and string format it as int?

Reply:
fixed 👍